backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/serialization/KonanIrlinker.kt

```diff
@@ -105,7 +105,12 @@ internal class KonanIrLinker(
             return KonanInteropModuleDeserializer(moduleDescriptor, isCached)
         }
 
-        return KonanModuleDeserializer(moduleDescriptor, klib ?: error("Expecting kotlin library"), strategy)
+        // Deserialize only bodies for inline functions.
+        val optimalStrategy = if (klib is KotlinLibrary && cachedLibraries.isLibraryCached(klib))
```

Comment:
See
https://github.com/JetBrains/kotlin-native/blob/1b6d2bb2b046c1ec5a6c569776a5179957483640/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/PsiToIr.kt#L120 
It's better to add another branch there and use `KotlinIrLinker.deserializeHeadersWithInlineBodies`.

Reply:
Fixed